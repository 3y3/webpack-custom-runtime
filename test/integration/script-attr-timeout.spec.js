'use strict';

const cct = require('./utils/custom-configurable-test');
const WebpackCustomRuntime = require('../../').CustomRuntimePlugin;

describe('ScriptAttr(Timeout)', function() {

    function cit(name, options, config) {
        cct(name, Object.assign({
            plugins: [
                new WebpackCustomRuntime({ behavior: '' }),
                new WebpackCustomRuntime.ScriptAttr.Timeout(options),
            ]
        }, config));
    }

    cit('should not to override script timeout by default', {});

    cit('should override script timeout from options', { value: 50 });

    cit('should override script type from config', {}, {
        output: {
            chunkLoadTimeout: 60000
        }
    });
});
