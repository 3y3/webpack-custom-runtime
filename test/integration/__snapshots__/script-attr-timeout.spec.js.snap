// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScriptAttr(Timeout) should not to override script timeout by default: 0.js 1`] = `
"Index: 0.js
===================================================================
--- 0.js	
+++ 0.js	
"
`;

exports[`ScriptAttr(Timeout) should not to override script timeout by default: 1.js 1`] = `
"Index: 1.js
===================================================================
--- 1.js	
+++ 1.js	
"
`;

exports[`ScriptAttr(Timeout) should not to override script timeout by default: main-module.js 1`] = `
"Index: main-module.js
===================================================================
--- main-module.js	
+++ main-module.js	
"
`;

exports[`ScriptAttr(Timeout) should not to override script timeout by default: other-module.js 1`] = `
"Index: other-module.js
===================================================================
--- other-module.js	
+++ other-module.js	
"
`;

exports[`ScriptAttr(Timeout) should not to override script timeout by default: runtime.js 1`] = `
"Index: runtime.js
===================================================================
--- runtime.js	
+++ runtime.js	
@@ -150,0 +150,2 @@
+/******/ 	  result.timeout = 120;
+/******/
"
`;

exports[`ScriptAttr(Timeout) should override script timeout from options: 0.js 1`] = `
"Index: 0.js
===================================================================
--- 0.js	
+++ 0.js	
"
`;

exports[`ScriptAttr(Timeout) should override script timeout from options: 1.js 1`] = `
"Index: 1.js
===================================================================
--- 1.js	
+++ 1.js	
"
`;

exports[`ScriptAttr(Timeout) should override script timeout from options: main-module.js 1`] = `
"Index: main-module.js
===================================================================
--- main-module.js	
+++ main-module.js	
"
`;

exports[`ScriptAttr(Timeout) should override script timeout from options: other-module.js 1`] = `
"Index: other-module.js
===================================================================
--- other-module.js	
+++ other-module.js	
"
`;

exports[`ScriptAttr(Timeout) should override script timeout from options: runtime.js 1`] = `
"Index: runtime.js
===================================================================
--- runtime.js	
+++ runtime.js	
@@ -150,0 +150,2 @@
+/******/ 	  result.timeout = 50;
+/******/
"
`;

exports[`ScriptAttr(Timeout) should override script type from config: 0.js 1`] = `
"Index: 0.js
===================================================================
--- 0.js	
+++ 0.js	
"
`;

exports[`ScriptAttr(Timeout) should override script type from config: 1.js 1`] = `
"Index: 1.js
===================================================================
--- 1.js	
+++ 1.js	
"
`;

exports[`ScriptAttr(Timeout) should override script type from config: main-module.js 1`] = `
"Index: main-module.js
===================================================================
--- main-module.js	
+++ main-module.js	
"
`;

exports[`ScriptAttr(Timeout) should override script type from config: other-module.js 1`] = `
"Index: other-module.js
===================================================================
--- other-module.js	
+++ other-module.js	
"
`;

exports[`ScriptAttr(Timeout) should override script type from config: runtime.js 1`] = `
"Index: runtime.js
===================================================================
--- runtime.js	
+++ runtime.js	
@@ -150,0 +150,2 @@
+/******/ 	  result.timeout = 60;
+/******/
"
`;
